### DISCUSSION TITLE
Welcome to coral_usb_ros Discussions!

### DISCUSSION THREAD

**Discussion by knorth55** [jsk-ros-pkg/coral_usb_ros#71] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by slabban** [comment]
Hello!

I'm Ros Developer and System Engineer by profession, at heart I'm a roboticist with a passion for Perception. A quick link to my portfolio here:
https://www.slabban.dev/

I'm currently working with the Edge TPU, where, at a high level, I will use this ros package to run on four camera's simultaneously.   
As you may conclude, I ran into the issue where launching four instances of this node does not work since one node will block the device port.

My current workaround is to create an intermediate node that subscribes to the four raw image topics and concatenates the images together using opencv and output them as one sensor_msgs/Image topic that would be fed into this package. I figure this would be a quick solve to the port issue and would lower the computational processing on the TPU.
I'm curious to know if any of you have tried this approach or have found nice workarounds.

My other curiosity is how you were able to process scenarios where both the class topic and bounding box topics were needed. My current approach is to use the exact time policy to tie the topics together and process them in on e callback function.

Cheers,
Sam

**Reply by knorth55** [reply]
actually, one node does not block the port, but the model is too huge that one device cannot run multiple devices on one device.
So the solution is using multiple coral tpu usb.
There is also multi tpu PCIe board from asus, which can run 8 tpu in one board.

**Reply by knorth55** [reply]
And if you want to use bounding box and classification result, you need to use message filter.

**Reply by slabban** [reply]
That's greatly insightful. Since the edgu TPUs are currently out of stock, I will try to concatenate the images and use the summed imaged as input to the node, will let you know how it goes!
